pyiceberg/io/pyarrow.py

```diff
@@ -1194,6 +1194,7 @@ def _task_to_record_batches(
     case_sensitive: bool,
     name_mapping: Optional[NameMapping] = None,
     use_large_types: bool = True,
+    limit: Optional[int] = None,
```

Comment:
I like the #1043 way where `_task_to_record_batches` is kept as is; its just generating more batches. The function which calls `_task_to_record_batches` can keep track of the `limit`

Reply:
"technically" we can also push down the limit to the `ds.Scanner.from_fragment`'s `batch_size` parameter, but that's currently out of scope 
https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html#pyarrow.dataset.Scanner.from_fragment